### PR TITLE
@k8slens/logger

### LINKED ISSUE #7662: Extract logger into own package

### PR DESCRIPTION
**Pull request by gabriel-mirantis**

### CODE REVIEW

File: packages/core/src/extensions/as-legacy-globals-for-extension-api/as-legacy-global-singleton-object-for-extension-api.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { asLegacyGlobalForExtensionApi, getLegacyGlobalDiForExtensionApi } from "@k8slens/legacy-global-di";
+import { loggerInjectionToken } from "@k8slens/logger";
+
+export interface LegacySingleton<T> {
+  createInstance(): T;
+  getInstance(): T;
+  resetInstance(): void;
+}
+
+export function asLegacyGlobalSingletonForExtensionApi<Instance>(injectable: Injectable<Instance, unknown, void>): LegacySingleton<Instance>;
```
Comment (Nokel81) [inline packages/core/src/extensions/as-legacy-globals-for-extension-api/as-legacy-global-singleton-object-for-extension-api.ts]: Why is this needed? Should this be part of `@k8slens/legacy-global-di`?

Comment (gabriel-mirantis) [reply packages/core/src/extensions/as-legacy-globals-for-extension-api/as-legacy-global-singleton-object-for-extension-api.ts]: this was an effect of the merge and lint-fix

**Review by Nokel81 — APPROVED** [review]
